packages/@aws-cdk/aws-lambda/test/test.function-hash.ts

```diff
@@ -0,0 +1,132 @@
+      test.deepEqual(trimFromStart('hello', 4), 'ello');
+      test.deepEqual(trimFromStart('hello', 1), 'o');
+      test.done();
+    }
+
+  },
+
+  "calcHash": {
+    'same configuration and code yields the same hash'(test: Test) {
+      const stack1 = new Stack();
+      const fn1 = new lambda.Function(stack1, 'MyFunction1', {
+        runtime: lambda.Runtime.NODEJS_12_X,
+        code: lambda.Code.fromAsset(path.join(__dirname, 'handler.zip')),
+        handler: 'index.handler'
+      });
+
+      const stack2 = new Stack();
+      const fn2 = new lambda.Function(stack2, 'MyFunction1', {
+        runtime: lambda.Runtime.NODEJS_12_X,
+        code: lambda.Code.fromAsset(path.join(__dirname, 'handler.zip')),
+        handler: 'index.handler'
+      });
+
+      test.deepEqual(calculateFunctionHash(fn1), calculateFunctionHash(fn2));
+      test.deepEqual(calculateFunctionHash(fn1), 'aea5463dba236007afe91d2832b3c836');
+      test.done();
+    },
+  },
+
+  'code impacts hash'(test: Test) {
+    const stack1 = new Stack();
+    const fn1 = new lambda.Function(stack1, 'MyFunction1', {
+      runtime: lambda.Runtime.NODEJS_12_X,
+      code: lambda.Code.fromAsset(path.join(__dirname, 'my-lambda-handler')),
+      handler: 'index.handler'
+    });
+
+    test.notDeepEqual(calculateFunctionHash(fn1), 'aea5463dba236007afe91d2832b3c836');
+    test.deepEqual(calculateFunctionHash(fn1), '979b4a14c6f174c745cdbcd1036cf844');
+    test.done();
+  },
+
+  'environment variables impact hash'(test: Test) {
+    const stack1 = new Stack();
+    const fn1 = new lambda.Function(stack1, 'MyFunction', {
+      runtime: lambda.Runtime.NODEJS_12_X,
+      code: lambda.Code.fromAsset(path.join(__dirname, 'my-lambda-handler')),
+      handler: 'index.handler',
+      environment: {
+        Foo: 'bar'
+      }
+    });
+
+    const stack2 = new Stack();
+    const fn2 = new lambda.Function(stack2, 'MyFunction', {
+      runtime: lambda.Runtime.NODEJS_12_X,
+      code: lambda.Code.fromAsset(path.join(__dirname, 'my-lambda-handler')),
+      handler: 'index.handler',
+      environment: {
+        Foo: 'beer'
+      }
+    });
+
+    test.deepEqual(calculateFunctionHash(fn1), 'd1bc824ac5022b7d62d8b12dbae6580c');
+    test.deepEqual(calculateFunctionHash(fn2), '3b683d05465012b0aa9c4ff53b32f014');
+    test.done();
+  },
+
+  'runtime impacts hash'(test: Test) {
+    const stack1 = new Stack();
+    const fn1 = new lambda.Function(stack1, 'MyFunction', {
+      runtime: lambda.Runtime.NODEJS_12_X,
+      code: lambda.Code.fromAsset(path.join(__dirname, 'my-lambda-handler')),
+      handler: 'index.handler',
+      environment: {
+        Foo: 'bar'
+      }
+    });
+
+    const stack2 = new Stack();
+    const fn2 = new lambda.Function(stack2, 'MyFunction', {
+      runtime: lambda.Runtime.NODEJS_10_X,
+      code: lambda.Code.fromAsset(path.join(__dirname, 'my-lambda-handler')),
+      handler: 'index.handler',
+      environment: {
+        Foo: 'beer'
+      }
```

Comment:
this code is fine, but it would be nice to add a regression test for flipping the environment variables around to make sure that use case is validated.

e.g. 

```
environment: {
    Foo: 'beer',
    Bar: "deer"
}
...
environment: {
    Bar: "deer",
    Foo: 'beer',
}
```

produce the same hash.

Reply:
Good catch!